src/backend/SyncBackendConnection.cpp

```diff
@@ -695,6 +681,65 @@ std::vector<StatisticsData> SyncBackendConnection::get_data(
     }
 }
 
+bool SyncBackendConnection::data_available(
+        DataKind data_kind,
+        EntityId source_entity_id,
+        EntityId target_entity_id,
+        Timestamp start_time /* = Timestamp() */,
+        Timestamp end_time /* = std::chrono::system_clock::now() */)
+{
+    // TODO: Replace by the actual function in the backend when it is implemented.
+    std::vector<StatisticsData> data = get_data(
+        data_kind, source_entity_id, target_entity_id, 1, StatisticKind::NONE, start_time, end_time);
+    return !data.empty();
+}
+
+bool SyncBackendConnection::build_source_target_entities_vectors(
+        DataKind data_kind,
+        EntityId source_entity_id,
+        EntityId target_entity_id,
+        std::vector<EntityId>& source_ids,
+        std::vector<EntityId>& target_ids)
+{
+    bool two_entities_data = false;
+    source_ids.clear();
+    target_ids.clear();
+
+    try
+    {
+        for (auto kinds_supported : StatisticsBackend::get_data_supported_entity_kinds(data_kind))
+        {
+            // Get the entities of the kind required for the data type that are related with the entity source
+            auto source_ids_tmp = get_entities(kinds_supported.first, source_entity_id);
+            source_ids.insert(source_ids.end(), source_ids_tmp.begin(), source_ids_tmp.end());
+
+            // Get the entities of the kind required for the data type that are related with the entity target
+            if (kinds_supported.second != EntityKind::INVALID)
+            {
+                // If the target entity is required but not given, the source is used
+                // This is usefule for two entities DataKinds when want to ask for all the targets for a source entity
+                // Useful for summary
+                two_entities_data = true;
+                if (!target_entity_id.is_valid())
+                {
+                    target_entity_id = source_entity_id;
+                }
+                auto target_ids_tmp = get_entities(kinds_supported.second, target_entity_id);
+                target_ids.insert(target_ids.end(), target_ids_tmp.begin(), target_ids_tmp.end());
+            }
+        }
+    }
+    catch (const Exception& e)
+    {
+        qWarning() << "Fail getting data: " << e.what();
+        static_cast<void>(e); // In release qWarning does not compile and so e is not used
+
+        return false;
+    }
```

Comment:
This exception does not need to be handled here, it is done in get_data.

Reply:
Yes, this is needed as there is a call to a backend function (`StatisticsBackend::get_data_supported_entity_kinds()`). 
However, I've modified the logging message.